### ISSUE TITLE
Better Templates

### ISSUE THREAD

**Issue by rehoyt** [LibreHealthIO/LibreEHR#661]
It would be ideal if there were more customizable templates, as opposed to Layout Forms which are labor intensive and non-intuitive

**Comment by aethelwulffe** [comment]
Does Template mean "Clinical Encounter Form"?
I know the drop-and-drag form generator I have been working on for about three years still needs about 400 more hours of work, and as I earn my living doing fabrication and design contracting (this unfortunately relegated to hobby status), that time is very difficult to find.  I really wish I could bring it forward,  but it looks like another two years.  🙁

**Comment by aethelwulffe** [comment]
Forms with hidden sections however, are much easier....provided there is a blueprint to start with.

**Comment by rehoyt** [comment]
I am speaking specifically about template-driven encounter notes/forms written with a SOAP format. 
A great template would be customizable with 1. Coded diagnoses 2. Customizable history with drop down options 3. Links to database table elements. For example if the template was for diabetes, it would automatically link to tables for  the last weight, BMI, blood glucose, A1c, Blood pressure and LDL cholesterol

**Comment by aethelwulffe** [comment]
You mean make a form that queries for various diagnosis, then provides review and input fields based on current Dx?

**Comment by rehoyt** [comment]
It would be organized into Subjective, Objective, Assessment and Plan (SOAP). For diabetes patients there would be a set number of questions in each section with each question being customizable. Here is an example of how Practice Fusion does it. Not very fancy as no information is automatically inserted into the note/template
Screen Shot 2017-08-10 at 11.42.39 AM

**Comment by aethelwulffe** [comment]
Screenshot did not come up.
I would note that the SOAP format seems to not be very popular with clinicians these days.  It also lends itself to narrative usage instead of empirical points.  In a pure text setting, it makes perfect sense, not so much for data-driven information.
As a DISPLAY format, there may be some value to organizing the sections in such a way.
Try to get the screenshot of what you are talking about posted again.  That will help us come to an intercept of concept!

**Comment by rehoyt** [comment]
I was searching for open source software for EHR templates. Nothing jumped out at me but I did see a [website](http://www.ackdoc.com/fpot.html) with lots of common disease templates in a MS Word format. Is there any reasonable way to digitize these forms and make them customizable? Seems like some of the information could be mapped to the database, such as vital signs, etc. 
HL7 has a template working group but looks like they were using the HL7 v. 3 RIM model which is not useful. Seems like FHIR calls for resources (templates) might be a better solution

**Comment by aethelwulffe** [comment]
Yes.  Open them up (preferably in LibreOffice), then save them as a .html file.  Post them somewhere.  We can turn them into forms from there....or we can just grab them from whereever.  These might be good for Code-In students to chew on.

**Comment by tmccormi** [comment]
Most "Templates" use bad layout that does not take into account the data (like Vitals) that have already been collected. Or worse the are a pile of check boxes and descision trees that lack the freedom needed to actually work with the patient.   I'm pretty sure that NLP from a progress note in conjunction with some preformatted "text snippets" is a way better than a bunch of predesigned forms.

**Comment by aethelwulffe** [comment]
Actually, yes, these word documents stink.  I was thinking that they had data input fields and things like that (actual template documents) but they are just text thingies (except for some of the fancier pdf versions).  As text sources for making a real form(making lists, etc...), they might be of help.  Turning them into integrated empirical clinical-data gathering user interfaces is a whole different scale of logic vs. the effort that were put into any of these.

**Comment by tmccormi** [comment]
Projects like CTakes already purport to to be able to do this.  I should be able to demonstrate that next week when Hadrian gets back in town (I've already seen a first demo).

I did not like the implementation of Nation Notes, but I did like the concept of a set of forms that included predefined and user definable snippets that included "Fill in" tags.   It's a way better model that separate templates/forms for each type of problem.

**Comment by tmccormi** [comment]
Here's a video demo of it openEMR for Bob ..
https://www.youtube.com/watch?v=5cRmhg0D7Qw

**Comment by rehoyt** [comment]
When the beta testers have finished I'll be sure to ask them about the value of templated notes for students. I don't mind setting up 4-5 templates for common disorders. I assume it is easy to add Nation Notes to LibreHealth??
